File: golang/4-findMedianSortedArrays.go
package golang

func FindMedianSortedArrays(nums1 []int, nums2 []int) float64 {
	mixedList := nums1
	tempIndex := 0
	for i, num := range nums2 {
		for tempIndex != len(mixedList) {
			if mixedList[tempIndex] >= num {
				mixedList = append(mixedList, 0)
				copy(mixedList[tempIndex+1:], mixedList[tempIndex:])
				mixedList[tempIndex] = num
				tempIndex += 1
				break
			}
			tempIndex += 1
		}
		if tempIndex == len(mixedList) {
			mixedList = append(mixedList, nums2[i:]...)
			break
		}
	}
	index := len(mixedList)
	if index%2 == 1 {
		return float64(mixedList[index/2])
	} else {
		return float64(mixedList[index/2]+mixedList[index/2-1]) / 2
	}
}

File: ts/4-findMedianSortedArrays.ts
export const findMedianSortedArrays = (
  nums1: number[],
  nums2: number[],
): number => {
  let mixedList = nums1
  let tempIndex = 0
  nums2.some((num, index) => {
    while (tempIndex !== mixedList.length) {
      if (mixedList[tempIndex] >= num) {
        mixedList.splice(tempIndex, 0, num)
        tempIndex += 1
        break
      }
      tempIndex += 1
    }
    const isEnd = tempIndex === mixedList.length
    if (isEnd) {
      mixedList = mixedList.concat(nums2.slice(index))
    }
    return isEnd
  })
  const centerIndex = mixedList.length / 2
  return centerIndex % 1 === 0
    ? (mixedList[centerIndex] + mixedList[centerIndex - 1]) / 2
    : mixedList[centerIndex - 0.5]
}
